core/deployment/src/main/java/io/quarkus/deployment/recording/PropertyUtils.java

```diff
@@ -22,9 +22,32 @@ public Property[] apply(Class<?> type) {
             List<Property> ret = new ArrayList<>();
             Method[] methods = type.getMethods();
 
+            List<Method> checkedMethodList = new ArrayList<>();
+
+            for (Method method : methods) {
+                if (method.getName().startsWith("get")) {
+                    checkedMethodList.add(method);
+                } else if (method.getName().startsWith("is")) {
+                    boolean hasDuplicate = false;
+                    for (Method maybeDuplicate : methods) {
+                        // prefer get method over is method
+                        if (maybeDuplicate.getName().equals("get" + method.getName().substring(2))) {
+                            hasDuplicate = true;
+                            break;
+                        }
+                    }
+
+                    if (!hasDuplicate) {
+                        checkedMethodList.add(method);
+                    }
+                } else if (method.getName().startsWith("set")) {
+                    checkedMethodList.add(method);
+                }
+            }
+
             Map<String, Method> getters = new HashMap<>();
             Map<String, Method> setters = new HashMap<>();
```

Comment:
I thought a bit more about it and I think you should have three maps here and add a `isGetters` map.
You would then at the end of the loop, remove everything from the `isGetters` map that has a key in `getters` (`isGetters.keySet().removeAll(getters.keySet())` should do the trick). And then as a last step, you can add the `isGetters` content to `getters`.

That should simplify things a bit and would avoid going through the loop twice doing mostly the same thing.

Reply:
+1 on the map; looking at how `getters` is used, I think it would be more clear to change
```
Method get = getters.get(i);
```

to

```
Method get = getters.get(i);
if (get == null) {
    get = isGetters.get(i); // If there is no "get" getter, use the "is" getter
}
```

(which in turn remove the need for removing the "get" getter keys from the "isGetters" map).

Reply:
Works for me :).